lib/Modules/PathPatternMatcher.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Tawk\Modules;
+
+use Tawk\Enums\WildcardLocation;
+use Tawk\Helpers\PathHelper;
+
+class PathPatternMatcher {
+	/**
+	 * Matches current path to multiple patterns
+	 */
+	public static function match($current_path_chunks, $pattern_paths_chunks) {
+		foreach($pattern_paths_chunks as $pattern_path_chunks) {
+			$wildcard_loc = PathHelper::get_wildcard_location_by_chunks($pattern_path_chunks);
+
+			if ($wildcard_loc === WildcardLocation::NONE) {
+				if (join('/', $current_path_chunks) === join('/', $pattern_path_chunks)) {
+					return true;
+				};
+			} else if ($wildcard_loc === WildcardLocation::START) {
+				// if wildcard is at the start, match in reverse order so that the wildcard is at the end
+				if (self::match_chunks_reverse($current_path_chunks, $pattern_path_chunks) === true) {
+					return true;
+				}
+			} else {
+				if (self::match_chunks($current_path_chunks, $pattern_path_chunks) === true) {
+					return true;
+				}
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Iterates over the current and pattern path chunks and matches them
+	 */
+	private static function match_chunks($current_path_chunks, $pattern_path_chunks) {
+		$wildcard_loc = PathHelper::get_wildcard_location_by_chunks($pattern_path_chunks);
```

Comment:
We're getting the location in here and in `match` which seems wasteful. Could provide `stop_on_wildcard` parameter as `$wildcard_loc === WildcardLocation::MIDDLE`